pkg/antctl/transform/networkpolicy/transform.go

```diff
@@ -17,50 +17,122 @@ package networkpolicy
 import (
 	"io"
 	"reflect"
+	"sort"
 	"strconv"
 
 	"github.com/vmware-tanzu/antrea/pkg/antctl/transform"
 	"github.com/vmware-tanzu/antrea/pkg/antctl/transform/common"
 	cpv1beta "github.com/vmware-tanzu/antrea/pkg/apis/controlplane/v1beta2"
+	"github.com/vmware-tanzu/antrea/pkg/controller/networkpolicy"
 )
 
 type Response struct {
 	*cpv1beta.NetworkPolicy
 }
 
-func objectTransform(o interface{}) (interface{}, error) {
+func objectTransform(o interface{}, _ map[string]string) (interface{}, error) {
 	return Response{o.(*cpv1beta.NetworkPolicy)}, nil
 }
 
-func listTransform(l interface{}) (interface{}, error) {
+func listTransform(l interface{}, opts map[string]string) (interface{}, error) {
 	policyList := l.(*cpv1beta.NetworkPolicyList)
+	sortBy := ""
+	if sb, ok := opts["sort-by"]; ok {
+		sortBy = sb
+	}
+	npSorter := &NPSorter{
+		networkPolicies: policyList.Items,
+		sortBy:          sortBy,
+	}
+	sort.Sort(npSorter)
 	result := make([]Response, 0, len(policyList.Items))
-	for i := range policyList.Items {
-		o, _ := objectTransform(&policyList.Items[i])
+	for i := range npSorter.networkPolicies {
+		o, _ := objectTransform(&npSorter.networkPolicies[i], opts)
 		result = append(result, o.(Response))
 	}
 	return result, nil
 }
 
-func Transform(reader io.Reader, single bool) (interface{}, error) {
+func Transform(reader io.Reader, single bool, opts map[string]string) (interface{}, error) {
 	return transform.GenericFactory(
 		reflect.TypeOf(cpv1beta.NetworkPolicy{}),
 		reflect.TypeOf(cpv1beta.NetworkPolicyList{}),
 		objectTransform,
 		listTransform,
+		opts,
 	)(reader, single)
 }
 
+var (
+	sortByEffectivePriority = "effectivePriority"
+	// Compute a tierPriority value in between the application tier and the baseline tier,
+	// which can be used to sort all policies by tier.
+	effectiveTierPriorityK8sNP = (networkpolicy.DefaultTierPriority + networkpolicy.BaselineTierPriority) / 2
+)
+
+type NPSorter struct {
+	networkPolicies []cpv1beta.NetworkPolicy
+	sortBy          string
+}
+
+func (nps *NPSorter) Len() int { return len(nps.networkPolicies) }
+func (nps *NPSorter) Swap(i, j int) {
+	nps.networkPolicies[i], nps.networkPolicies[j] = nps.networkPolicies[j], nps.networkPolicies[i]
+}
+func (nps *NPSorter) Less(i, j int) bool {
+	switch nps.sortBy {
+	case sortByEffectivePriority:
+		var ti, tj int32
+		if nps.networkPolicies[i].TierPriority == nil {
+			ti = effectiveTierPriorityK8sNP
+		} else {
+			ti = *nps.networkPolicies[i].TierPriority
+		}
+		if nps.networkPolicies[j].TierPriority == nil {
+			tj = effectiveTierPriorityK8sNP
+		} else {
+			tj = *nps.networkPolicies[j].TierPriority
+		}
+		if ti != tj {
+			return ti < tj
+		}
+		pi, pj := nps.networkPolicies[i].Priority, nps.networkPolicies[j].Priority
+		if pi != nil && pj != nil && *pi != *pj {
+			return *pi < *pj
+		}
+		fallthrough
+	default:
```

Comment:
I didn't see any place where we check that `--sort-by` is either not provided or set to `effectivePriority`?
I didn't check manually, but we should make sure that there is a good error message if someone uses `--sort-by=.spec.priority` or some other unsupported value. We should not just default to the name / UID in that case.

Reply:
Added that check in handler.